rudder-core/src/main/scala/com/normation/rudder/domain/queries/CmdbQuery.scala

```diff
@@ -410,28 +410,35 @@ case object AgentComparator extends CriterionType {
    * We are using 'oldShortName' because we are matching on substring,
    * and that name was used in the past.
    */
-  val agentTypes = ( (ANY_CFENGINE, "Any CFEngine based agent") :: AgentType.allValues.toList.map(a => (a.oldShortName, a.displayName))).sortBy( _._2 )
+  val agentTypes = ( (ANY_CFENGINE, "Any CFEngine based agent") :: AgentType.allValues.toList.map(a => (a.id, a.displayName))).sortBy( _._2 )
 
+  val agentMap = AgentType.allValues.toList.map(a => (a.id, a :: Nil)).toMap
```

Comment:
I believe you didn't do what you intented here: agentMap does not contain ANY_CFENGINE, I believe you wanted to do something like: 
- delete agentTypes
- val agentMap = ( (ANY_CFENGINE, "Any CFEngine based agent") :: AgentType.allValues.toList.map(a => (a.id, a :: Nil))).toMap

Reply:
Was on first version then eclipse havoc ... adding it back